lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -347,9 +460,17 @@ void SessionStateEnforcer::m5g_update_session_state_to_amf(
                  << " with teid " << teid;
     return;
   }
-  bool amf_update_pending = false;
-  uint32_t cur_version;
-  auto& session       = **session_it;
+  auto& session = **session_it;
+  cur_version   = session->get_current_version();
+  if (version < cur_version) {
+    MLOG(MINFO) << "UPF verions of session imsi " << imsi << " of  teid "
+                << teid << " recevied version " << version
+                << " SMF latest version: " << cur_version << " Resending";
+    if (session->inc_throttle()) {
+      m5g_send_session_request_to_upf(imsi, session);
+    }
+    return;
+  }
   auto session_update = SessionStore::get_default_session_update(session_map);
   SessionStateUpdateCriteria& session_uc =
       session_update[imsi][session->get_session_id()];

```

Comment:
Nit: You could just load session_id into a variable so you don't have to keep calling the function.

Reply:
adopted.